hooks/node_handlers/base_export_handler.py

```diff
@@ -0,0 +1,644 @@
+            index = len(entries)
+        else:
+            index = entries.index(version_str)
+        sgtk_version.set(index)
+
+    def _get_optional_fields(self, node, template):
+        """
+        Get the optional fields from the sgtk_optional_keys parm.
+
+        :param node: A :class:`hou.Node` instance.
+        :param template: An :class:`sgtk.Template` instance.
+
+        :rtype: dict
+        """
+        parm = node.parm(self.OPTIONAL_KEYS)
+        if not parm:
+            return {}
+        return json.loads(parm.evalAsString())
+
+    def _populate_from_file_path(self, node, file_path):
+        """
+        Populate a node's sgtk parms from a file path.
+        If the file path doesn't match the template, disable the
+        use of shotgun on this node.
+
+        :param node: A :class:`hou.Node` instance.
+        :param str file_path: The file path to populate from.
+        """
+        template = self._get_template_for_file_path(node, file_path)
+        skip_keys = self.get_optional_keys(template)
+        fields = template.validate_and_get_fields(file_path, skip_keys=skip_keys)
+        if not fields:
+            use_sgtk = node.parm(self.USE_SGTK)
+            use_sgtk.set(False)
+            self._enable_sgtk(node, False)
+            output_parm = node.parm(self.OUTPUT_PARM)
+            output_parm.set(file_path)
+        else:
+            fields.update(self._get_optional_fields(node, template))
+            fields["SEQ"] = "FORMAT: $F"
+            current_version = fields.get("version", self.NEXT_VERSION_STR)
+            all_versions = self._resolve_all_versions_from_fields(node, fields, template)
+            self._populate_from_fields(node, fields)
+            self._update_all_versions(node, all_versions)
+            self._set_version(node, current_version)
+            self._refresh_file_path(node, update_version=False)
+
+    def _restore_sgtk_parms(self, node):
+        """
+        Restore any removed sgtk parameters onto the given node.
+
+        :param node: A :class:`hou.Node` instance containing sgtk parameters.
+        """
+        output_parm = node.parm(self.OUTPUT_PARM)
+        original_file_path = output_parm.unexpandedString()
+        self.add_sgtk_parms(node)
+        self._populate_from_file_path(node, original_file_path)
+
+    def _get_sequence_paths(self, path, template, fields):
+        """
+        Get all the paths relating to a sequence.
+
+        :param str path: The sequence path.
+        :param template: An :class:`sgtk.Template`.
+        :param dict fields: The template fields.
+         
+        :returns: A list of file paths
+        """
+        fields = copy.deepcopy(fields)
+        if "SEQ" in fields:
+            fields["SEQ"] = "FORMAT: $F"
+            path = template.apply_fields(fields)
+            glob_path = self._get_sequence_glob_path(path, template, fields)
+            sequence_paths = glob.glob(glob_path)
+            return sequence_paths
```

Comment:
Maybe force return type to always be list

```suggestion
        sequence_paths = []
        if "SEQ" in fields:
            fields["SEQ"] = "FORMAT: $F"
            path = template.apply_fields(fields)
            glob_path = self._get_sequence_glob_path(path, template, fields)
            sequence_paths = glob.glob(glob_path)
        return sequence_paths
```